src/Files.App/Actions/FileSystem/CreateAlternateDataStreamAction.cs

```diff
@@ -0,0 +1,115 @@
+	{
+		private readonly IContentPageContext context;
+
+		private static readonly IFoldersSettingsService FoldersSettingsService = Ioc.Default.GetRequiredService<IFoldersSettingsService>();
+
+		public string Label
+			=> "CreateAlternateDataStream".GetLocalizedResource();
+
+		public string Description
+			=> "CreateAlternateDataStreamDescription".GetLocalizedResource();
+
+		public override bool IsExecutable =>
+			context.HasSelection &&
+			context.CanCreateItem &&
+			(context.ShellPage?.ShellViewModel.WorkingDirectory != Path.GetPathRoot(context.ShellPage?.ShellViewModel.WorkingDirectory)) &&
+			UIHelpers.CanShowDialog;
+
+		public CreateAlternateDataStreamAction()
+		{
+			context = Ioc.Default.GetRequiredService<IContentPageContext>();
+
+			context.PropertyChanged += Context_PropertyChanged;
+		}
+
+		public async Task ExecuteAsync(object? parameter = null)
+		{
+			var nameDialog = DynamicDialogFactory.GetFor_CreateAlternateDataStreamDialog();
+			await nameDialog.TryShowAsync();
+
+			if (nameDialog.DynamicResult != DynamicDialogResult.Primary)
+				return;
+
+			var userInput = nameDialog.ViewModel.AdditionalData as string;
+			await Task.WhenAll(context.SelectedItems.Select(async selectedItem =>
+			{
+				var isDateOk = Win32Helper.GetFileDateModified(selectedItem.ItemPath, out var dateModified);
+				var isReadOnly = Win32Helper.HasFileAttribute(selectedItem.ItemPath, System.IO.FileAttributes.ReadOnly);
+
+				// Unset read-only attribute (#7534)
+				if (isReadOnly)
+					Win32Helper.UnsetFileAttribute(selectedItem.ItemPath, System.IO.FileAttributes.ReadOnly);
+
+				if (!Win32Helper.WriteStringToFile($"{selectedItem.ItemPath}:{userInput}", ""))
+				{
+					var dialog = new ContentDialog
+					{
+						Title = Strings.ErrorCreatingDataStreamTitle.GetLocalizedResource(),
+						Content = Strings.ErrorCreatingDataStreamDescription.GetLocalizedResource(),
+						PrimaryButtonText = "Ok".GetLocalizedResource()
+					};
+
+					if (ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 8))
+						dialog.XamlRoot = MainWindow.Instance.Content.XamlRoot;
+
+					await dialog.TryShowAsync();
+				}
+
+				// Restore read-only attribute (#7534)
+				if (isReadOnly)
+					Win32Helper.SetFileAttribute(selectedItem.ItemPath, System.IO.FileAttributes.ReadOnly);
+
+				// Restore date modified
+				if (isDateOk)
+					Win32Helper.SetFileDateModified(selectedItem.ItemPath, dateModified);
+			}));
+
+			if (context.ShellPage is null)
+				return;
+
+			if (FoldersSettingsService.AreAlternateStreamsVisible)
+				await context.ShellPage.Refresh_Click();
+			else
+			{
+				var dialog = new ContentDialog
+				{
+					Title = Strings.DataStreamsAreHiddenTitle.GetLocalizedResource(),
+					Content = Strings.DataStreamsAreHiddenDescription.GetLocalizedResource(),
+					PrimaryButtonText = Strings.Yes.GetLocalizedResource(),
+					SecondaryButtonText = Strings.No.GetLocalizedResource()
+				};
```

Comment:
Isn't it necessary to add an option to not notify again?

Reply:
Resolved in [801a858](https://github.com/files-community/Files/pull/16438/commits/801a8588357d8147779c6bc4db50aae7ed1f7496)